src/components/Configure/fields/OptionalFields/setOptionalField.ts

```diff
@@ -8,17 +8,22 @@ function setOptionalFieldProducer(
   fieldKey: string,
   checked: boolean,
 ) {
-  const draftSelectedOptionalFields = draft?.selectedOptionalFields || {};
+  const draftSelectedOptionalFields = draft?.read?.selectedOptionalFields || {};
   draftSelectedOptionalFields[fieldKey] = checked;
   if (!checked) {
     delete draftSelectedOptionalFields[fieldKey];
   }
-  const savedFields = draft.savedConfig.optionalFields;
-  const updatedFields = draftSelectedOptionalFields;
-  const isModified = !isFieldObjectEqual(savedFields, updatedFields);
-  // immer exception if we try to set a value
-  // eslint-disable-next-line no-param-reassign
-  draft.isOptionalFieldsModified = isModified;
+
+  if (draft.read?.savedConfig.optionalFields) {
+    const savedFields = draft.read.savedConfig.optionalFields;
+    const updatedFields = draftSelectedOptionalFields;
+    const isModified = !isFieldObjectEqual(savedFields, updatedFields);
+    // immer exception if we try to set a value
+    // eslint-disable-next-line no-param-reassign
+    draft.read.isOptionalFieldsModified = isModified;
+  } else {
+    console.warn('read.savedConfig.optionalFields is undefined');
```

Comment:
is this going to always log for integrations that don't have optional fields? if so, that might be kind of annoying.

Reply:
This code should not be reachable. I put this here for debugging purposes, if this becomes a case that I need to look at in the future.